internal/ingestion/knowledge_compile/reader.go

```
@@ -53,6 +53,22 @@ type Reader interface {
+// loads one stable dataset-level page by id; keeping it separate from Reader
+// avoids forcing test/offline readers and non-Wiki variants to implement a
+// dataset-row lookup they do not need.
+type mergedProductReader interface {
+	LoadMergedProduct(ctx context.Context, tenant, kb, id string) (kccommon.Product, error)
+}
+
+type mergedWikiPageReader interface {
+	LoadMergedWikiPages(ctx context.Context, tenant, kb string) ([]kccommon.Product, error)
+}
+
+type mergedWikiPageDeleter interface {
```

golangci-lint (2.12.2)

[error] type mergedWikiPageDeleter is unused

(unused)